### PR TITLE
Ocs provisioning api checking lists

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
Added some functions to be able to check lists of users and groups.

### CODE REVIEW

File: build/integration/features/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -59,6 +58,50 @@ public function getOCSResponse($response){
 	}
 
 	/**
+    *  Parses the xml answer to get the array of users returned.
+    */
+    public function getArrayOfUsersResponded($resp) {
```
Comment (DeepDiver1975) [inline build/integration/features/bootstrap/FeatureContext.php]: indent seems off - which IDE are you using? PHPStorm is suggested - THX

Comment (MorrisJobke) [inline build/integration/features/bootstrap/FeatureContext.php]: I think the intendation here is in spaces and not in tabs. Please use tabs instead

File: build/integration/features/provisioning-v1.feature [hunk 2]
```diff
@@ -40,6 +40,13 @@ Feature: provisioning
     And the HTTP status code should be "200"
 
 
+  Scenario: Getting all users
+    Given As an "admin"
+    When sending "GET" to "/cloud/users"
+    And users returned are
+      | brand-new-user | admin |
```
Comment (DeepDiver1975) [inline build/integration/features/provisioning-v1.feature]: this implies that these users have previously been created - scenarios should be self-contained.
Please add a
````
 Given the user "brand-new-user" exists
 And the user "admin" exists

File: build/integration/features/provisioning-v1.feature [hunk 3]
```diff
@@ -74,6 +81,13 @@ Feature: provisioning
     And group "new-group" exists
 
 
+  Scenario: Getting all users
+    Given As an "admin"
+    When sending "GET" to "/cloud/groups"
+    And groups returned are
+      | admin | new-group |
```
Comment (DeepDiver1975) [inline build/integration/features/provisioning-v1.feature]: same as above

Comment (DeepDiver1975) [inline build/integration/features/provisioning-v1.feature]: in addition - the pipe separated strings is acutally table - is should look like

````
| admin |
| new-group |

File: build/integration/features/bootstrap/FeatureContext.php [hunk 4]
```diff
@@ -59,6 +58,50 @@ public function getOCSResponse($response){
...
+    }
+
+	/**
+	 * @Then /^users returned are$/
+	 * @param \Behat\Gherkin\Node\TableNode|null $formData
+	 */
+	public function theUsersShouldBe($usersList) {
+		if ($usersList instanceof \Behat\Gherkin\Node\TableNode) {
+			$users = $usersList->getRows()[0];
+            $respondedArray = $this->getArrayOfUsersResponded($this->response);
```
Comment (DeepDiver1975) [inline build/integration/features/bootstrap/FeatureContext.php]: please please - clean code -indent is totally  :boom: here - THX

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
@MorrisJobke please review - THX

**Comment by DeepDiver1975** [comment]
@SergioBertolinSG THX - let's move this in and continue with additional tests in a followup pr - THX

**Comment by MorrisJobke** [comment]
Looks good :+1:

**Pull request merged**